thing.go

```diff
@@ -43,6 +43,20 @@ type Thing struct {
 	// Thing using the data spectrum classifications from the ODI: Open, Shared
 	// or Closed
 	Visibility Visibility `json:"visibility"`
+
+	// ThingType indicates what type is this thing, it is short for thingful:isThingType
+	// the most generic thing type is thingful:ConnectedDevice
+	ThingType string `json:"thingType"`
+
+	// Category indicates what type this thing belongs to, it is short for thingful:hasCategory
+	// This is mainly used by thingful.net. Only values allowes are what is defined in thingfulx/category.go
+	// for example, thingfulx.Environment.Name
+	Category string `json:"category"`
+
```

Comment:
Is it safe to loose the CategoryUID here? I think pomelo is still using it when saving new things